bin/add-test-widgets-to-sidebar.php

```diff
@@ -0,0 +1,460 @@
+		'url'           => $media->guid,
+	);
+}
+
+/**
+ * Get the settings for the gallery widget.
+ *
+ * @return array $audio_widget The settings for the gallery widget.
+ */
+function amp_gallery_widget() {
+	$images = amp_media( 'image', 3 );
+	return array(
+		'ids' => wp_list_pluck( $images, 'ID' ),
+	);
+}
+
+/**
+ * Gets the settings for the image widget.
+ *
+ * @return array $image_widget The settings for the image widget.
+ */
+function amp_image_widget() {
+	$all_images        = amp_media( 'image', 1 );
+	$image             = reset( $all_images );
+	$metadata          = wp_get_attachment_metadata( $image->ID );
+	$default_dimension = 100;
+	return array(
+		'attachment_id' => $image->ID,
+		'height'        => isset( $metadata['height'] ) ? $metadata['height'] : $default_dimension,
+		'width'         => isset( $metadata['width'] ) ? $metadata['width'] : $default_dimension,
+		'url'           => wp_get_attachment_url( $image->ID ),
+	);
+}
+
+/**
+ * Get media items of a certain type.
+ *
+ * @throws Exception If not enough posts exist.
+ * @param  integer $type  The post_mime_type of the media item.
+ * @param  integer $count The number of images for which to query.
+ * @return array|WP_CLI::error The media IDs, or an error on failure.
+ */
+function amp_media( $type, $count = 3 ) {
+	$query = new \WP_Query(
+		array(
+			'post_type'      => 'attachment',
+			'post_mime_type' => $type,
+			'post_status'    => 'inherit',
+			'posts_per_page' => $count,
+		)
+	);
+	if ( $query->post_count < $count ) {
+		throw new Exception(
+			sprintf(
+				'Please ensure at least %1$s "%2$s" attachments are accessible and run this again. There are currently only %3$s.',
+				$count,
+				$type,
+				$query->found_posts
+			)
+		);
+	}
+	return $query->get_posts();
+}
+
+/**
+ * Gets a menu ID, if it has more than 4 items.
+ *
+ * Iterates through all of the menus.
+ * And returns the first that has 4 items.
+ *
+ * @throws Exception When there is no menu with 4 items.
+ * @return integer|WP_CLI::error $menu_id The menu ID, or an error if no menu has at least 4 items.
+ */
+function amp_menu() {
+	$menus         = wp_get_nav_menus();
+	$minimum_count = 4;
+	foreach ( $menus as $menu ) {
+		if ( $menu->count > $minimum_count ) {
+			return $menu->term_id;
+		}
+	}
+	throw new Exception( 'Please add at least 4 items to a menu.' );
+}
+
+/**
+ * Whether a widget with the same values exists in the sidebar.
+ *
+ * Iterate through all of the widgets in the sidebar.
+ * Find all of the widgets with the same ID base, like 'text.'
+ * Then, find if one of those has the same settings as the new widget settings.
+ *
+ * @param  array  $widget  The settings of the widget.
+ * @param  string $sidebar The slug of the sidebar.
+ * @return boolean $in_sidebar Whether the widget is in the sidebar.
+ */
+function amp_widget_already_in_sidebar( $widget, $sidebar ) {
+	$sidebars           = wp_get_sidebars_widgets();
+	$widgets_in_sidebar = $sidebars[ $sidebar ];
```

Comment:
Make sure this isset.

Reply:
Fixed [here](https://github.com/Automattic/amp-wp/pull/859/commits/dba75caef1efc20d9f9e88888069f11ead68bce8).